test.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/sh
+
+set -e
+set -u
+
+# Defaults
+operation=full
+noconfirm=0
+clean_all=0
+failure=1 # Assume test failure
+flags=""
+
+eecho () {
+	echo "$@" >&2
+}
+
+usage () {
+	eecho "$0 [flags] [operation]"
+	eecho
+	eecho "operation:"
+	eecho "  full (default)"
+	eecho "  clean"
+	eecho "  help (this help)"
+	eecho
+	eecho "flags:"
+	eecho "  -h: help (this help)"
+	eecho "  -d: turn on nlplug-findfs debug output"
+	eecho "  -y: skip confirmation at program start"
+	eecho "  -x: clean all (including produced binaries)"
+	eecho
+	exit 1
+}
```

Comment:
http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_07_04:

> If the redirection symbol is "<<-", all leading <tab> characters shall be stripped from input lines and the line containing the trailing delimiter.

That’s the reason why I use tabs in shell scripts, although I prefer spaces evewyhere else.